apps/extension/src/services/index.ts

```diff
@@ -99,6 +100,6 @@ export class Services {
     await ws.indexedDb.clear();
     await localExtStorage.set('lastBlockSynced', 0);
     this.walletServicesPromise = undefined;
-    await this.tryToSync();
+    await this.onServiceWorkerInit();
```

Comment:
If you do this, won't there me multiple listeners registered

```typescript
chrome.runtime.onMessage.addListener(swMessageHandler);
```

and cause a memory leak?

Reply:
did not pay attention to it! can i add `chrome.runtime.onMessage.removeListener(swMessageHandler);` to clearCache?

Reply:
I'd move `chrome.runtime.onMessage.addListener(swMessageHandler);` to `service-worker.ts` and rename `onServiceWorkerInit` to `initialize()`